File: src/api/categories.ts
import request from './request'
import { mockDb } from '@/mock/database'
import { useMock, unwrap } from './runtime'
import type { DeleteResult } from '@/types/api'
import type { Category, CategoryPayload } from '@/types/category'

export const getCategories = (includeDisabled = false): Promise<Category[]> => useMock
  ? mockDb.getCategories(includeDisabled)
  : request.get<Category[]>('/categories', { params: includeDisabled ? { include_disabled: true } : undefined }).then(unwrap)
export const createCategory = (data: CategoryPayload): Promise<Category> => useMock ? mockDb.createCategory(data) : request.post<Category>('/categories', data).then(unwrap)
export const updateCategory = (id: number, data: CategoryPayload): Promise<Category> => useMock ? mockDb.updateCategory(id, data) : request.put<Category>(`/categories/${id}`, data).then(unwrap)
export const updateCategoryStatus = (id: number, enabled: boolean): Promise<Category> => useMock ? mockDb.updateCategoryStatus(id, enabled) : request.patch<Category>(`/categories/${id}/status`, { enabled }).then(unwrap)
export const deleteCategory = async (id: number): Promise<DeleteResult> => useMock
  ? { deleted: await mockDb.deleteCategory(id) }
  : request.delete<DeleteResult>(`/categories/${id}`).then(unwrap)

File: backend/app/api/routes/categories.py
from fastapi import APIRouter, status

from app.api.deps import AdminUser, CurrentUser, DbSession
from app.schemas.category import CategoryPayload, CategoryRead, CategoryStatusUpdate
from app.schemas.common import ApiResponse
from app.services.category_service import CategoryService


router = APIRouter(prefix="/categories", tags=["Categories"])


@router.get("", response_model=ApiResponse[list[CategoryRead]], summary="分类列表")
def list_categories(db: DbSession, current_user: CurrentUser, include_disabled: bool = False) -> ApiResponse[list[CategoryRead]]:
    return ApiResponse(data=CategoryService.list(db, current_user, include_disabled=include_disabled))


@router.post("", response_model=ApiResponse[CategoryRead], status_code=status.HTTP_201_CREATED, summary="新增分类")
def create_category(payload: CategoryPayload, db: DbSession, admin: AdminUser) -> ApiResponse[CategoryRead]:
    return ApiResponse(data=CategoryService.create(db, payload, admin), message="分类已创建")


@router.put("/{category_id}", response_model=ApiResponse[CategoryRead], summary="更新分类")
def update_category(category_id: int, payload: CategoryPayload, db: DbSession, admin: AdminUser) -> ApiResponse[CategoryRead]:
    return ApiResponse(data=CategoryService.update(db, category_id, payload, admin), message="分类已更新")


@router.patch("/{category_id}/status", response_model=ApiResponse[CategoryRead], summary="启用或禁用分类")
def update_category_status(category_id: int, payload: CategoryStatusUpdate, db: DbSession, admin: AdminUser) -> ApiResponse[CategoryRead]:
    return ApiResponse(data=CategoryService.update_status(db, category_id, payload, admin), message="分类状态已更新")


@router.delete("/{category_id}", response_model=ApiResponse[dict[str, bool]], summary="删除分类")
def delete_category(category_id: int, db: DbSession, admin: AdminUser) -> ApiResponse[dict[str, bool]]:
    CategoryService.delete(db, category_id, admin)
    return ApiResponse(data={"deleted": True}, message="分类已删除")
